asap/rough_align/fit_multiple_solves.py

```diff
@@ -0,0 +1,367 @@
+    "affine_output_stack": dict(example_render, **{
+            "name": "MN12_L2_1A_affine_test", #use your own name
+            "collection_type": "stack",
+            "db_interface": "render"
+            }), #output of affine transform
+    "thin_plate_output_stack": dict(example_render, **{
+            "name": "Mn12_L2_1A_tps_test", #use your own name
+            "collection_type": "stack",
+            "db_interface": "render"
+            }),# output of thin plate spline transform
+    "minZ":0, #first remapped Z value as an integer
+    "maxZ": 801, #last remapped Z value as an integer
+    "pool_size": 20, #customizable. 20 is probably overkill
+    "close_stack": False,
+    "output_json":"/path/to/output/json/files" #required for proper functioning of the module.
+}
+
+logger = logging.getLogger()
+
+class ApplyMultipleSolvesException(RenderModuleException):
+    """Raise exception by using try except blocks...."""
+class TileException(RenderModuleException):
+    """Raise this when unmatched tile ids exist or if multiple sections per z value exist"""
+
+def create_resolved_tiles(render,pm_render,input_stack: str,pm_collection: str,solve_range: tuple): #input_stack and pm_collection are names of the (a) montage-scaped & Z-mapped stack, and (b) pointmatch collection from render, respectively; solve_range is a tuple of the minimum and maximum of the new_z values.
+    
+    stack_zs = [z for z in renderapi.stack.get_z_values_for_stack(input_stack, render=render) if solve_range[0] <= z <= solve_range[-1]]
+    
+    num_threads = 35 #customizable
+    with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as e:
+        rtfut_to_z = {e.submit(renderapi.resolvedtiles.get_resolved_tiles_from_z, input_stack, z, render=render): z for z in stack_zs}
+        z_to_rts = {rtfut_to_z[fut]: fut.result() for fut in concurrent.futures.as_completed(rtfut_to_z)}
+    
+    matchgroups = {ts.layout.sectionId for ts in (i for l in (rts.tilespecs for rts in z_to_rts.values()) for i in l)}
+    with concurrent.futures.ThreadPoolExecutor(max_workers=num_threads) as e:
+        futs = [e.submit(renderapi.pointmatch.get_matches_with_group, pm_collection, g, render=pm_render) for g in matchgroups]
+        allmatches = [i for l in (fut.result() for fut in concurrent.futures.as_completed(futs)) for i in l]
+
+    match_pIds, match_qIds = map(set, zip(*[(m["pId"], m["qId"]) for m in allmatches]))
+    match_tileIds = match_pIds | match_qIds
+    all_tileIds = {ts.tileId for ts in (i for l in (rts.tilespecs for rts in z_to_rts.values()) for i in l)}
+
+    tIds_nomatch = all_tileIds - match_tileIds  # any tiles without matches
+    
+    return {'resolved_tiles': z_to_rts,'matched_tiles': match_tileIds,'All_matches': allmatches,'All_tiles': all_tileIds,'Unmatched_tiles': tIds_nomatch}
+
+def multiz_unmatchz_counter(z_to_rts,combined_rts,tIds_nomatch):
```

Comment:
These should be separate checks and the multiz check should be optional.
There are use cases where we expect multiple sections on a single z, e.g. sections split during pickup.